core/shared/src/main/scala/scalaz/zio/Schedule.scala

```diff
@@ -109,7 +132,7 @@ trait Schedule[-A, +B] { self =>
    * Returns a new schedule that continues this schedule so long as the predicate
    * is satisfied on the output value of the schedule.
    */
-  final def whileValue(f: B => Boolean): Schedule[A, B] =
+  final def whileOutput(f: B => Boolean): Schedule[A, B] =
```

Comment:
How about `outputWhile`? It seems `<verb><conjunction>` is more idiomatic. In `IO`: `retryOrElse`, `seqWith`, `sandboxWith`, etc. Also reminds me of `takeWhile`, `takeUntil` et al. the vernacular of traversable data structures.

Reply:
Here's the meaning of these things:

 * `whileInput` — it decides to continue for as long as the input matches the specified condition
 * `untilInput` — it decides to continue until the input matches the specified condition
 * `whileOutput` — it decides to continue for as long as the output matches the specified condition
 * `untilOutput` — it decides to continue until the output matches the specified condition

It's not so much that it will "output" while the condition is satisfied, but that the schedule will continue for as long the output of the schedule satisfies the condition.

This suggests terminology like `continueWhileOutput` or something (although that's too long). I'm not sure.

Reply:
ymmv, but I'm ok with some methods having long names if they are specific to a fairly niche use case. Examples like `traverseWithKey`, `mapKeysMonotonic` for maps in haskell come to mind. In this case though, I think you're right that it's too long, as these look like valuable combinators a user would typically embed in larger expressions.